edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -86,9 +100,29 @@ internal static CreateContainerParameters GetCreateOptions(string createOptions)
         }
 
         internal static string ValidateAndGetImage(string image)
+        {
+            return ValidateAndGetImage(image, new EnvironmentWrapper());
+        }
+
+        internal static string ValidateAndGetImage(string image, IEnvironmentWrapper env)
         {
             image = Preconditions.CheckNonWhiteSpace(image, nameof(image)).Trim();
+
+            Match match_host = ImageUpstreamRegex.Match(image);
+            if ((match_host.Success)
+                && (match_host.Groups["host"]?.Length > 0)
+                && (match_host.Groups["path"]?.Length > 0))
+            {
+                String host_address = env.GetVariable(match_host.Groups["host"].Value);
+
+                if (host_address != null)
```

Comment:
if it is null, you don't change image variable and let it continue to original code path.  shouldn't we return fail immediately?

Reply:
you need to check null or white space

Reply:
I'll add a check